crossing-training.lic

```diff
@@ -382,15 +382,36 @@ class CrossingTraining
   def calculate_mana(min, more, discern_data, cyclic)
     total = min + more
     total = (total * @settings.prep_scaling_factor).floor
-    discern_data['mana'] = [(total / 4.0).ceil, min].max
+    discern_data['mana'] = [(total / 5.0).ceil, min].max
     remaining = total - discern_data['mana']
-    if remaining > @settings.cambrinth_cap
+    total_cambrinth_mana = 0
+    @settings.cambrinth_items.each do |item|
+      total_cambrinth_mana += item[:cap]
+    end
+    if remaining > total_cambrinth_mana && !!@settings.cambrinth_items
+      discern_data['mana'] = discern_data['mana'] + (remaining - @settings.cambrinth_cap)
+      remaining = total - discern_data['mana']
+    elsif remaining > @settings.cambrinth_cap && !@settings.cambrinth_items
       discern_data['mana'] = discern_data['mana'] + (remaining - @settings.cambrinth_cap)
       remaining = total - discern_data['mana']
     end
     if cyclic
       discern_data['cambrinth'] = nil
       discern_data['mana'] = discern_data['mana'] + remaining
+    elsif remaining > 0 && !!@settings.cambrinth_items
+      discern_data['cambrinth'] = []
+      @settings.cambrinth_items.each do |item|
+        step_size = (item[:cap]/2).floor
+        if remaining > step_size
+          discern_data['cambrinth'] << [remaining, step_size].min
```

Comment:
If you wanted to reduce code duplication you could use a loop: `2.times do { ... }`